### PR TITLE
DCGAN

### PR DESCRIPTION
**Pull request by hungjinh**
develop logs for adding dcgan networks

### CODE REVIEW

**Comment by hungjinh** [comment]
**dcgan_generator.py** future development works (after the v0 pipeline): 

- [x] Don't save image arrays into `self.trainInfo['img_list']` (file gets too large). Output image directly instead. 

- [x]  Increase the output image frequency.
- [ ] Add `lr_scheduler` to control the learning rate.
- [ ] Allow `lr` to be save in `self.trainInfo`.
